# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/ProxyObject.cpp
# Commit: 22854b710fbf
# Full Hash: 22854b710fbfb779ac6b19fe4c9ac2570189cec0
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 6 - Move Realm* from ObjectGroup to BaseShape. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106976
# ==============================================================================

diff -r 4531d5c25694 -r 22854b710fbf js/src/vm/ProxyObject.cpp
--- a/js/src/vm/ProxyObject.cpp	Fri Mar 05 19:10:06 2021 +0000
+++ b/js/src/vm/ProxyObject.cpp	Fri Mar 05 19:10:07 2021 +0000
@@ -101,7 +101,8 @@
       return nullptr;
     }
 
-    shape = EmptyShape::getInitialShape(cx, clasp, proto, /* nfixed = */ 0);
+    shape =
+        EmptyShape::getInitialShape(cx, clasp, realm, proto, /* nfixed = */ 0);
     if (!shape) {
       return nullptr;
     }
@@ -109,8 +110,7 @@
     realm->newProxyCache.add(group, shape);
   }
 
-  MOZ_ASSERT(group->realm() == realm);
-  MOZ_ASSERT(shape->zone() == cx->zone());
+  MOZ_ASSERT(shape->realm() == realm);
   MOZ_ASSERT(!IsAboutToBeFinalizedUnbarriered(group.address()));
   MOZ_ASSERT(!IsAboutToBeFinalizedUnbarriered(shape.address()));
 